src/sonic-yang-mgmt/_sonic_yang_ext.py

```diff
@@ -0,0 +1,616 @@
+
+    return True
+
+"""
+load JSON schema format from yang models
+"""
+def loadJsonYangModel(self):
+
+    for f in self.yangFiles:
+        m = self.ctx.get_module(f)
+        if m is not None:
+            xml = m.print_mem(ly.LYD_JSON, ly.LYP_FORMAT)
+            self.yJson.append(parse(xml))
+
+    return
+
+"""
+Create a map from config DB tables to container in yang model
+"""
+def createDBTableToModuleMap(self):
+
+    for j in self.yJson:
+        # get module name
+        moduleName = j['module']['@name']
+        if "sonic-head" in moduleName or "sonic-common" in moduleName:
+            continue;
+        # get all top level container
+        topLevelContainer = j['module']['container']
+        if topLevelContainer is None:
+            raise Exception("topLevelContainer not found")
+
+        assert topLevelContainer['@name'] == moduleName
+
+        container = topLevelContainer['container']
+        # container is a list
+        if isinstance(container, list):
+            for c in container:
+                self.confDbYangMap[c['@name']] = {
+                    "module" : moduleName,
+                    "topLevelContainer": topLevelContainer['@name'],
+                    "container": c
+                    }
+        # container is a dict
+        else:
+            self.confDbYangMap[container['@name']] = {
+                "module" : moduleName,
+                "topLevelContainer": topLevelContainer['@name'],
+                "container": container
+                }
+    return
+
+"""
+Get module, topLevelContainer and json container for a config DB table
+"""
+def get_module_top_container(self, table):
+    cmap = self.confDbYangMap
+    m = cmap[table]['module']
+    t = cmap[table]['topLevelContainer']
+    c = cmap[table]['container']
+    return m, t, c
+
+"""
+Crop config as per yang models
+"""
+def cropConfigDB(self, croppedFile=None):
+
+    for table in self.jIn.keys():
+        if table not in self.confDbYangMap:
+            del self.jIn[table]
+
+    if croppedFile:
+        with open(croppedFile, 'w') as f:
+            dump(self.jIn, f, indent=4)
+
+    return
+
+"""
+Extract keys from table entry in Config DB and return in a dict
+"""
+def extractKey(self, tableKey, regex):
+
+    # get the keys from regex of key extractor
+    keyList = re.findall(r'<(.*?)>', regex)
+    # create a regex to get values from tableKey
+    # and change separator to text in regexV
+    regexV = re.sub('<.*?>', '(.*?)', regex)
+    regexV = re.sub('\|', '\\|', regexV)
+    # get the value groups
+    value = re.match(r'^'+regexV+'$', tableKey)
+    # create the keyDict
+    i = 1
+    keyDict = dict()
+    for k in keyList:
+        if value.group(i):
+            keyDict[k] = value.group(i)
+        else:
+            raise Exception("Value not found for {} in {}".format(k, tableKey))
+        i = i + 1
+
+    return keyDict
+
+"""
+Fill the dict based on leaf as a list or dict @model yang model object
+"""
```

Comment:
Is this function used for filling out default values?   Where is this function used?    Fill the data that are not in config_db.json but in yang models with default values?    All fields in yang model(s) or just the mandatory ones?

Reply:
If you have a Yang data file generated/xslated from config_db.json,   will you be able to add default values and all data to the Yang data json file and load the Yang data json file in one shot?

Reply:
This function is called from 
xlateList(self, model, yang, config, table)---> createLeafDict(self, model):. 

We are creating map from leaf_name(key in configDB) to yang Leaf for yang list. such as below for scope. This map is used to easily find attributes of each key from config DB in YANG LIST i.e to find Yang leaf.
"scope": {
                           "type": {
                              "enum": [
                                 {
                                    "_name": "global"
                                 },
                                 {
                                    "_name": "local"
                                 }
                              ],
                              "_name": "enumeration"
                           },
                           "_name": "scope"
                        },

Reply:
> If you have a Yang data file generated/xslated from config_db.json, will you be able to add default values and all data to the Yang data json file and load the Yang data json file in one shot?

I did not get it ??
Default values are YANG properties, so Libyang adds them today, if needed. Yes, If a field is mandatory, user must enter it.

What you mean by 'one shot' here. You suspect data file will be so large that we will not be able to load it in single shot, if that is the case, even if we load in multiple shot, data tree will be huge. I did not get this point as of now?